# Introduction
Together with my classmate Rutger Bles, we had a look into a dataset to predict office occupancy based on factors like temperature, humidity, light, CO2, and humidity-ratio. We both used different models to check the accuracy of this predictive model. Where he used the KNN-model and I used the Logistic Regression Model.

## Business Understanding
Rutger and I used a dataset to predict office occupancy based on the following factors: temperature, humidity, light, CO2, and humidity-ratio.

## Import libraries
First load the necassary libraries.

In [1]:
import pandas as pd
from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import classification_report, accuracy_score

## Data Understanding
Load the dataset and explore it.

In [2]:
# Read the data
url = "https://raw.githubusercontent.com/HAN-M3DM-Data-Mining/assignments/master/datasets/KNN-occupancy.csv"
data = pd.read_csv(url)


## Data Preparation
Prepare the dataset for the logistic regression model. The data set is already clean.

In [3]:
# Define features (X) and target (y)
X = data[['Temperature', 'Humidity', 'Light', 'CO2', 'HumidityRatio']]
y = data['Occupancy']

# Split the data into training and testing sets (80% train, 20% test)
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

## Modeling
Training thelogistic regression model

In [4]:
# Create a logistic regression model
model = LogisticRegression(max_iter=1000)
model.fit(X_train, y_train)

# Evaluate the model
y_pred = model.predict(X_test)

## Evaluation and Deployment
Check the accuracy of the model that has just been trained.

In [5]:
# Calculate accuracy
accuracy = accuracy_score(y_test, y_pred)

# Generate classification report
report = classification_report(y_test, y_pred)

print(f'Accuracy: {accuracy}')
print('Classification Report:')
print(report)


Accuracy: 0.9877225291589933
Classification Report:
              precision    recall  f1-score   support

           0       1.00      0.99      0.99      1286
           1       0.95      1.00      0.97       343

    accuracy                           0.99      1629
   macro avg       0.97      0.99      0.98      1629
weighted avg       0.99      0.99      0.99      1629



The logistic regression model achieved impressive performance with an overall accuracy of 98.77% showing its strong predictive abilities.

# Conclusion
In conclusion, our task centered around training both a KNN-model and a Logistic Regression model for occupancy prediction. The results were exceptionally promising, boasting nearly flawless accuracy rates, accompanied by an impressively low count of false positives and false negatives. This underscores the remarkable reliability of the dataset and the remarkable predictive capabilities exhibited by our models.